### PR TITLE
Fix: clean up new ZMConversation message sender warning

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Clean up warnings after added deprecated warning to `ZMConversationMessage.sender` property.

### CODE REVIEW

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 1]
```diff
@@ -38,6 +38,13 @@
             ReferencedContainer = "container:Wire-iOS.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
+      <AdditionalOptions>
```
Comment (typfel) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme:41]: Revert this?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationTableViewDataSource.swift [hunk 2]
```diff
@@ -467,7 +467,7 @@ extension ConversationTableViewDataSource {
             !Message.isKnock(message) else { return false }
         
         guard let previousMessage = messagePrevious(to: message, at: index),
-            previousMessage.sender == message.sender,
+            previousMessage.senderUser as? ZMUser == message.senderUser as? ZMUser,
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationTableViewDataSource.swift:470]: Maybe you use the `===` operator instead and avoid the casting? https://www.hackingwithswift.com/example-code/language/whats-the-difference-between-equalsequals-and-equalsequalsequals

**Review by typfel — CHANGES_REQUESTED** [review]

**Review by KaterinaWire — APPROVED** [review]

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 3]
```diff
@@ -38,6 +38,13 @@
...
+         <AdditionalOption
+            key = "MallocStackLogging"
+            value = ""
+            isEnabled = "YES">
+         </AdditionalOption>
+      </AdditionalOptions>
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme:47]:
```suggestion
 
```

File: Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme [hunk 4]
```diff
@@ -38,6 +38,7 @@
             ReferencedContainer = "container:Wire-iOS.xcodeproj">
          </BuildableReference>
       </MacroExpansion>
+
```
Comment (billypchan) [inline Wire-iOS.xcodeproj/xcshareddata/xcschemes/Wire-iOS.xcscheme:41]:
```suggestion
```

**Review by typfel — APPROVED** [review]

**Pull request merged**